### PR TITLE
Feat(docs): Add doc for line ending errors

### PR DESCRIPTION
**Pull request by jonkaonor**
**Problem**
Errors would sometimes occur during the bookbrainz-site installation process (specifically when running develop.sh and database-init-docker.sh) if Linux line endings were replaced by Windows line endings in certain script files. This was occurring on Windows machines due to a default git setting that would replace all line endings in cloned files with Window's line endings (CRLF).

**Solution**
The main source of this error was addressed in https://github.com/metabrainz/bookbrainz-site/commit/b7b3a805eaa624687f02599ce3f37b7cf242393e  which changed the .gitattributes file in the bookbrainz-site repository to override the git setting. However this error may still occur if an application besides git changes line endings, therefore this commit adds documentation in the Troubleshooting section that gives step-by-step instructions on how to resolve the issue by changing git configuration settings and re-cloning the bookbrainz-site repository.

### CODE REVIEW

**Comment by jonkaonor** [comment]
Sorry I just realized you were asking for me to write this documentation on the installation-troubleshooting document in the bookbrainz-site repository and not this one.  I'll add the documentation to that shortly, feel free to close this request if this isn't the proper place for it.

**Comment by kellnerd** [comment]
I think you've submitted this PR in the right place, the plan is to move the docs from bookbrainz-site into this repo: https://github.com/metabrainz/bookbrainz-dev-docs/pull/11#issuecomment-1400173944

**Comment by jonkaonor** [comment]
Ah okay I read through the comments and understand now. Thank you for the clarification kellnerd.

File: docs/troubleshooting.rst [hunk 1]
```diff
@@ -32,6 +32,32 @@ General
...
+
+     git config --global core.autocrlf input
+
+  2. Delete the bookbrainz-site directory from its parent directory.
+  ::
+
+  3. Clone the bookbrainz repository again using the appropriate command (the URL will be different if cloning from a fork)
+  ::
+
+     git clone --recurse-submodules https://github.com/metabrainz/bookbrainz-site.git
```
Comment (MonkeyDo) [inline docs/troubleshooting.rst]: I think here we should just point to the relevant section in the docs.
The main reason is that when we copy instructions in multiple places it is easier to forget to modify all of them.
As a great example, we don't use submodules anymore so the `--recurse-submodules` flag hasn't been needed for a couple of years, and we forgot to update it everywhere :p

So I think we should have something like this:
```suggestion
  3. Clone the bookbrainz repository again using the appropriate command. Follow the instructions here: :ref:`cloning-repository`
```

Which requires adding this line above the "Cloning" header in [installation.rst](https://github.com/metabrainz/bookbrainz-dev-docs/blob/master/docs/installation.rst#cloning) (and an emtpty line in-between) to be able to reference it:
```
.. _cloning-repository:

Cloning
*******
```

File: docs/troubleshooting.rst [hunk 2]
```diff
@@ -32,6 +32,32 @@ General
...
+  by following these steps:
+
+  1. Navigate to the bookbrainz-site directory and configure git to clone repositories using the original line endings
+  with the following command:
+  ::
+
+     git config --global core.autocrlf input
+
+  2. Delete the bookbrainz-site directory from its parent directory.
+  ::
```
Comment (MonkeyDo) [inline docs/troubleshooting.rst]: This `::` is used to present a block of code or a command, and is not required here
```suggestion

```

File: docs/troubleshooting.rst [hunk 3]
```diff
@@ -32,6 +32,32 @@ General
...
+
+  3. Clone the bookbrainz repository again using the appropriate command (the URL will be different if cloning from a fork)
+  ::
+
+     git clone --recurse-submodules https://github.com/metabrainz/bookbrainz-site.git
+
+  4. The line endings should now be corrected and you may proceed with following the normal installation procedure.
+  ::
```
Comment (MonkeyDo) [inline docs/troubleshooting.rst]: Same as number 2.
```suggestion

```

**Review by MonkeyDo — CHANGES_REQUESTED** [review]
Thanks for adding this troubleshooting !

Indeed this is the right place for this PR :)

**Review by MonkeyDo — APPROVED** [review]
Took the liberty of finishing off this PR

**Pull request merged**